src/axios/cart/requests.js

```diff
@@ -0,0 +1,75 @@
+import { getUserId } from '../../helpers/functions';
+import axiosCart from './axios-cart';
+
+// existing requests for cart API
+
+const withCredentials = (productId) => {
+  const userId = getUserId();
+  const url = productId
+    ? `users/${userId}/cart/${productId}/.json`
+    : `users/${userId}/cart.json`;
+
+  return url;
+};
+
+export const fetchCartItems = async () => {
+  try {
+    const response = await axiosCart.get(withCredentials());
+    const inCart = [];
+
+    if (!response.data) return inCart;
+
+    Object.keys(response.data).forEach((key) => {
+      inCart.push({
+        id: key,
+        ...response.data[key],
+      });
+    });
+
+    return inCart;
+  } catch (error) {
+    const errorInfo = { isError: true, errorMessage: error.message };
```

Comment:
`return { isError: true, errorMessage: error.message };` is enough

But this is an improper place to put exception handling. If you don't plan to handle request exception with additional logic (wrapping exception to object is not an additional logic) you can simply drop try / catch block in this function and catch it closer to user.

Reply:
improved in the next commit https://github.com/maxSedelnikov/react-shopping-app/pull/1/commits/24984c3c8b52e6eb5707ea4a4edfa8373dc5a402